docs/GlobalAPI.md

```diff
@@ -1050,7 +1050,7 @@ test.each(table)('table as a variable example', (a, b, expected, extra) => {
 
 #### Template literal
 
-If all values are of the same type, the template literal API will type the arguments correctly:
```

Comment:
The idea here was to start with a simple and correctly working example.

I find it strange showing how something does not work. If a feature does not work, why it is shipped? Just remove it.

It is just an option, but I think documentation should show how to use a library instead of focussing on what does not work.

Reply:
Also your change makes this feature work better. Why to say something does not work? Hey, look! This works like a charm. Use it!

Reply:
I'm not sure what part you're referring to but I simplified it!

If case you were talking about the last part ([2nd commit](https://github.com/jestjs/jest/pull/14920/commits/f4f8533c58fff281f4671acb3d537dba85f278db)), I chose to leave it here, but more as a pitfall. wdyt?